withReduxSaga/index.js

```diff
@@ -0,0 +1,45 @@
+import createSagaMiddleware, { END } from 'redux-saga';
+import withSaga from './withReduxSaga';
+
+export default (config, InnerComponent, actions) => {
+  const middlwareOptions = config.sagaMonitor ? { sagaMonitor: config.sagaMonitor } : {};
+  const sagaMiddleware = createSagaMiddleware(middlwareOptions);
+
+  const configureStore = config.configureStore(sagaMiddleware);
+
+  const STORE_KEY = '_NEXT_STORE_KEY_';
+  const existingStore = (global.window && global.window[STORE_KEY]);
+  const params = {
+    store: existingStore || {},
+  };
+  params.makeStore = {
+    createStore: (initialState, options) => {
+      params.store = configureStore(initialState, options);
```

Comment:
I missed this response, sorry.

I'll think about how to do this cleanly.

I'm also wondering if the withReduxSaga should call the wrapped component's `getInitialState` and return it.  Which brings up the next question, does it call it before or after the store dispatch?  Things to think about.